kafka/manager.go

```diff
@@ -276,21 +276,36 @@ func (m *Manager) MonitorConsumerLag(topicConsumers []apmqueue.TopicConsumer) (m
 					continue
 				}
 				for partition, lag := range partitions {
+					// Check for errors in lag calculation, start offset, or end offset.
+					// The franz-go library can set lag to the end offset value when:
+					// - End offset is valid (lag.End.Err == nil)
+					// - Start offset has an error (lag.Start.Err != nil)  
+					// - No commit exists (lag.Commit.At == -1)
+					// This results in incorrect lag reporting, so we must check all error fields.
+					var lagErr error
 					if lag.Err != nil {
-						if lag.Err == kerr.UnknownTopicOrPartition {
+						lagErr = lag.Err
+					} else if lag.Start.Err != nil {
+						lagErr = lag.Start.Err
+					} else if lag.End.Err != nil {
+						lagErr = lag.End.Err
+					}
```

Comment:
Not sure it is possible and important, but what if multiple of these errors are non nil? Do you think it'd be useful to track them all in the error message?